src/main/java/com/band/api/resolvers/Mutation.java

```diff
@@ -26,24 +33,57 @@ public Mutation(UserRepository userRepository){
      * @param name
      * @return the created User
      */
-    public User createUser(String email, String username, String password, String name){
+    public User createUser(String email, String username, String password, String name) throws CustomException{
+        User user = null;
+        CustomException ex = null;
 
-        BCryptPasswordEncoder bc = new BCryptPasswordEncoder();
-        String passwordHash = bc.encode(password);
-        User user = User.builder()
-                    .email(email)
+        try {
+            verifyUsernameEmail(username,email);
+            String passwordHash = encoder.encode(password);
+            user = User.builder()
+                    .emailSearch(email.toLowerCase())
+                    .emailDisplay(email)
                     .username(username)
                     .passwordHash(passwordHash)
                     .name(name)
                     .build();
-        try {
             userRepository.save(user);
         }
+        catch (DataAccessResourceFailureException e){
+            //Database Unavailable
+            ex = new CustomException("Database Unavailable");
+        }
+        catch (CustomException e){
```

Comment:
Talked offline about handling exceptions around the methods that can throw them to make it clearer what causes the exceptions.
Also talked about exception name. @TheLastKiwi to create a BaseGraphQL exception and then subclass for specific exceptions (ie InvalidInputException).